app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/util/EISHttpReader.scala

```diff
@@ -63,12 +64,17 @@ class EISHttpReader(
     if (is2xx(response.status)) Right(jsonAs[T](response.body))
     else Left(response)
 
-  private def jsonAs[T](body: String)(implicit reads: Reads[T],  tt: TypeTag[T]): T = {
+  private def jsonAs[T](body: String)(implicit reads: Reads[T], tt: TypeTag[T]): T = {
     Try(Json.parse(body).as[T]) match {
       case Success(obj) => obj
       case Failure(exception) => throw new RuntimeException(s"Response body could not be read as type ${typeOf[T]}", exception)
     }
   }
+
+  def removeControlDocumentReferences(errorMsg: String): String = {
```

Comment:
this could be put in a trait and the HttpRader can derived it. There is already a ResponseHandler trait that contains the extractIfSuccessful and jsonAs already (these are identical to this inside Httprteader). So the removeContrlDocumentReference could go into ResponseHandler, HttpReader could derive from ResponseHandler and we can remove the extractIfSuccessful, jsonAs and removeContrlDocumentReference functions from HttpReader

Reply:
done